### PR TITLE
[Refactor] Refactor Confirm Leave Overlay View Logic

### PR DESCRIPTION
**Pull request by ShaunaSong**
## Purpose

* Refactor confirm leave overlay logic into ViewModel

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

1. Join the Group Call/ Teams Meeting
2. Press the Hang Up Button
3. Check Confirm Leave Overlay Behaviour

## What to Check
Verify that the following are valid
* Confirm Leave Overlay Behaviour does not Change

### CODE REVIEW

**Review by iaulakh — APPROVED** [review]
tested

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/hangup/ConfirmLeaveOverlayViewModel.kt [hunk 1]
```diff
@@ -3,29 +3,30 @@
...
 import com.azure.android.communication.ui.redux.action.Action
 import com.azure.android.communication.ui.redux.action.CallingAction
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 
 internal class ConfirmLeaveOverlayViewModel(
     private val dispatch: (Action) -> Unit,
 ) {
-    private var isConfirmLeaveOverlayDisplayed: Int = View.INVISIBLE
+    private var shouldDisplayConfirmLeaveOverlayStateFlow = MutableStateFlow(false)
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/hangup/ConfirmLeaveOverlayViewModel.kt:14]: can we change this to val

**Review by phileo99 — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/hangup/ConfirmLeaveOverlayViewModel.kt (hunk 1 above)
Comment (ShaunaSong) [reply azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/hangup/ConfirmLeaveOverlayViewModel.kt]: Updated, thanks

**Review by iaulakh — APPROVED** [review]

**Review by phileo99 — DISMISSED** [review]
👍

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/CallingFragment.kt [hunk 2]
```diff
@@ -155,6 +155,6 @@ internal class CallingFragment(
     }
 
     private fun requestCallEnd() {
-        confirmLeaveOverlayView.showHangupOverlay()
+        viewModel.getConfirmLeaveOverlayViewModel().show()
```
Comment (pavelprystinka) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/fragment/calling/CallingFragment.kt:158]: How about we will create a method CallingViewModel.requestCallEnd() and rename  ConfirmLeaveOverlayViewModel.show() to ConfirmLeaveOverlayViewModel.requestExitConfirmation()

Then from here: CallingFragment.requestCallEnd() call viewModel.requestCallEnd() which will call confirmLeaveOverlayViewModel.requestExitConfirmation()

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**